src/main/java/com/exasol/sql/rendering/AbstractFragmentRenderer.java

```diff
@@ -58,7 +58,7 @@ protected void appendSpace() {
     }
 
     protected void appendCommaWhenNeeded(final Fragment fragment) {
-        if (this.lastVisited.getClass().equals(fragment.getClass())) {
+        if (this.lastVisited != null && lastVisited.getClass().equals(fragment.getClass())) {
```

Comment:
Shouldn't there already be a method for that exact check? This is a check that happens in other places too.

Reply:
I just had a null pointer in tests so I fixed the place.